src/main/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/ParseRequestService.java

```diff
@@ -55,14 +57,27 @@ public void validateUserProfiles(List<UserProfile> userProfiles, UserRequest use
             boolean isInvalid = false;
             if (CollectionUtils.isEmpty(userProfile.getBaseLocation())) {
                 log.error("The base location is not available for the userProfile {} ", userProfile.getId());
+                invalidUserProfiles.add(userProfile);
                 isInvalid = true;
             }
             if (CollectionUtils.isEmpty(userProfile.getWorkArea())) {
                 log.error("The work area is not available for the userProfile {} ", userProfile.getId());
+                invalidUserProfiles.add(userProfile);
                 isInvalid = true;
             }
+            long invalidServiceCodeCount = userProfile.getWorkArea().stream().filter(workArea ->
+                   (!workArea.getServiceCode().isEmpty() && !workArea
+                   .getServiceCode().equals("BFA1"))).count();
+            if (invalidServiceCodeCount > 0) {
+                log.error("The userProfile {} has {} invalid service code",
+                        userProfile.getId(), invalidServiceCodeCount);
+                invalidUserProfiles.add(userProfile);
+                isInvalid = true;
+            }
+
```

Comment:
we can't add hardcode the BFA1 condition here, currently we are supporting only BFA1 but later we would be adding more. So this should be taken care only at drool side without any code change.